src/utils/index.ts

```diff
@@ -757,29 +757,31 @@ export const filterChannelListParams = (params: GroupChannelListQuery, channel:
   return true;
 };
 
-export const binarySearch = (list: Array<number>, number: number): number => { // [100, 99, 98, 97, ...]
-  const pivot = Math.floor(list.length / 2);
-  if (list[pivot] === number) {
-    return pivot;
-  }
-  const leftList = list.slice(0, pivot);
-  const rightList = list.slice(pivot + 1, list.length);
-  if (list[pivot] > number) {
-    return pivot + 1 + (rightList.length === 0 ? 0 : binarySearch(rightList, number));
-  } else {
-    return (leftList.length === 0) ? pivot : binarySearch(leftList, number);
-  }
-};
 // This is required when channel is displayed on channel list by filter
-export const getChannelsWithUpsertedChannel = (channels: Array<GroupChannel>, channel: GroupChannel): Array<GroupChannel> => {
-  if (channels.some((ch: GroupChannel) => ch.url === channel?.url)) {
-    return channels.map((ch: GroupChannel) => (ch.url === channel?.url ? channel : ch));
+export const getChannelsWithUpsertedChannel = (
+  _channels: Array<GroupChannel>,
+  channel: GroupChannel,
+  order?: GroupChannelListOrder,
+): Array<GroupChannel> => {
+  const compareFunc = match(order)
+    .with(GroupChannelListOrder.CHANNEL_NAME_ALPHABETICAL, () => (
+      (a: GroupChannel, b: GroupChannel) => a.name.localeCompare(b.name)
+    ))
+    .with(GroupChannelListOrder.CHRONOLOGICAL, () => (
+      (a: GroupChannel, b: GroupChannel) => b.createdAt - a.createdAt
+    ))
+    .otherwise(() => (
+      (a: GroupChannel, b: GroupChannel) => (b.lastMessage?.createdAt ?? Number.MIN_SAFE_INTEGER) - (a.lastMessage?.createdAt ?? Number.MIN_SAFE_INTEGER)
```

Comment:
What is one or both of them has no message? In that case we have to compare as CHANNEL_NAME_ALPHABETICAL

Reply:
Your suggestion is reasonable, but this logic is used by server side (we can see the channel list ordered like this when fetching channel list)

Reply:
![image](https://github.com/sendbird/sendbird-uikit-react/assets/46333979/73cfa199-317d-40b5-9ae3-68fdd433ce62)
eg. The most recent channel is located last